lib/charms/opensearch/v0/opensearch_plugins.py

```diff
@@ -452,66 +503,141 @@ def name(self) -> str:
         return "opensearch-knn"
 
 
+class OpenSearchPluginBackupDataProvider(OpenSearchPluginDataProvider):
+    """Responsible to decide which data to use for the backup plugin.
+
+    Backups should check different relations depending on their role in the cluster:
+    * main orchestrator
+    * failover orchestrator
+    * other
+    """
+
+    def __init__(self, charm):
+        """Creates the OpenSearchPluginBackupDataProvider object."""
+        super().__init__(charm)
+        self._relation = None
+        if not self._charm.opensearch_peer_cm.deployment_desc():
+            # Temporary condition: we are waiting for CM to show up and define which type
+            # of cluster are we. Once we have that defined, then we will process.
+            raise OpenSearchPluginMissingConfigError("Missing deployment description in peer CM")
```

Comment:
how is this exception caught?

Reply:
It is caught on the main touch points of the `openseach_plugins`: the `self.plugin_manager.run()` and in backups, we catch it on `self.charm.plugin_manager.apply_config()`.